pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
If you haven't already, please read:
https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#spec-and-status (the spec part)
https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#defaulting
https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#late-initialization

"The specification is a complete description of the desired state, including configuration settings provided by the user, default values expanded by the system, and properties initialized or otherwise changed after creation by other ecosystem components (e.g., schedulers, auto-scalers), and is persisted in stable storage with the API object."

Examples of fields that are automatically set:
* Default values: pod.spec.restartPolicy, pod.spec.containers[].imagePullPolicy
* Admission control: pod.spec.containers[].resources.limits.cpu
* Late initialization: pod.spec.nodeName, service.spec.clusterIP
* Horizontal auto-scaling and rolling update: replicationcontroller.spec.replicas
* Vertical auto-sizing: pod.spec.containers[].resources.requests.cpu

Some of these fields are immutable once set, and there are readonly fields in the metadata that can't be set by the user, either.

Due to all of these processes, it is not possible for a user to edit their original config file and feed it to `kubectl replace`, which expects a fully specified resource, at least not without using `--force`, which is very disruptive, since it deletes and recreates the resource.

THAT is the workflow that I want to enable: edit your configs and do `kubectl apply -f myconfigs/`. Everything else, such as conflict detection, is gravy.

Everything not explicitly specified by the user in their configs should be reserved for operational tweaking, both manual (kubectl scale, kubectl label, kubectl patch, etc.) and automated (auto-scaling, auto-sizing, automated deployments, etc.).

As I mentioned elsewhere, the user should not be re-running kubectl apply just to ensure conformance of the live state to their desired state. If the live state diverges, it's either user error, such as misconfiguring an auto-scaler, or intentional, which is where conflict detection upon the next update they try to do would be nice.